main.js

```diff
@@ -0,0 +1,248 @@
+const key = '5dbc25bc8e1707e2fd6040f71c4df0dd';
+
+let city = localStorage.getItem('defaultCity') ? localStorage.getItem('defaultCity') : 'No default city';
+
+/** 
+ * Sets global city variable to city from geolocation
+ */
+
+const handleGeolocation = (data) => {
+  let arr = data.split(' ')
+  let address = arr[1]
+  const add1 = address.replace(',', '')
+  city = add1;
+};
+
+/**
+ * Uses google API to get the city based on users current coordinates
+ * @param {number} lat latitude
+ * @param {number} lon longitude
+ */
+
+const getGeolocation = (lat, lon) => {
+  let url = `https://maps.googleapis.com/maps/api/geocode/json?latlng=${lat},${lon}&key=AIzaSyB-4vMY0awy9BejmkOB8Az55XNs2-GJEiI`;
+
+  fetch(url, {
+    method: "GET",
+    dataType: "json"
+  })
+    .then(data => data.json())
+    .then(data => handleGeolocation(data.plus_code.compound_code))
+    .catch(error => console.error('Error fetching data:', error));
+};
+
+document.addEventListener('DOMContentLoaded', () => {
+  document.querySelector('.default-city').innerHTML = `Default city: ${city}`;
+  document.querySelector('.geolocation').addEventListener('click', () => {
+    navigator.geolocation.getCurrentPosition((pos) => handleClickGeo(pos));
+  });
+});
+
+/**
+ * Distributes lat and lon to getWeather and getFiveDay functions, and sets city variable through getGeolocation
+ * @param {object} pos position object from navigator.geolocation API
+ */
+
+const handleClickGeo = (pos) => {
+
+  const loader = '<div class="loader-container"><div class="loader"></div></div>';
+  document.querySelector('body').insertAdjacentHTML('beforebegin', loader);
+
+  let lat = pos.coords.latitude;
+  let lon = pos.coords.longitude;
+
+  getGeolocation(lat, lon);
+
+  let data = [{
```

Comment:
data isn't a very descriptive variable name - you can describe the type of data to make a better named variable